buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -878,6 +879,25 @@ public int getBytes(int index, GatheringByteChannel out, int length)
     }
 
     @Override
+    public int getBytes(int index, FileChannel out, int position, int length)
+            throws IOException {
+        int count = nioBufferCount();
+        if (count == 1) {
+            return out.write(internalNioBuffer(index, length), position);
+        } else {
+            long writtenBytes = 0;
+            for (ByteBuffer buf : nioBuffers(index, length)) {
+                writtenBytes += out.write(buf, position + writtenBytes);
```

Comment:
nvm...it seems `nioBuffers` should limit the total amount of data in the buffers to `length`